### PR TITLE
SCA with Php Inspections (EA Ultimate)

### PR DESCRIPTION
**Pull request by kalessil**
Minor code tweaks:
- dead code
- mkdir race condition
- sprintf pattern fix
- random API usage tweaks

### CODE REVIEW

File: eZ/Publish/Core/Base/ServiceContainer.php [hunk 1]
```diff
@@ -225,7 +225,7 @@ protected function dumpContainer(ConfigCache $cache)
     protected function prepareDirectory($directory, $name)
     {
         if (!is_dir($directory)) {
-            if (false === @mkdir($directory, 0777, true)) {
+            if (!@mkdir($directory, 0777, true) && !is_dir($directory)) {
```
Comment (glye) [inline eZ/Publish/Core/Base/ServiceContainer.php]: Could use an explanation here, please.

**Comment by glye** [comment]
I'm ok with merging as is.

File: eZ/Publish/Core/Base/ServiceContainer.php (hunk 1 above)
Comment (kalessil) [reply eZ/Publish/Core/Base/ServiceContainer.php]: Sure thing: https://github.com/kalessil/phpinspectionsea/blob/master/docs/probable-bugs.md#mkdir-race-condition =) Concurrency is fun, isn't it?

**Comment by kalessil** [comment]
Thank you @glye =)

**Pull request merged**